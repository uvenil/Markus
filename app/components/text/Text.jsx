'use strict';

import React from 'react';

export default class Text extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const theme     = this.props.theme === 'dark' ? require('../../theme.dark.json') : require('../../theme.light.json');
        const textColor = this.props.textColor ? this.props.textColor : theme.primaryTextColor;

        return (
            <span
                className={this.props.className}
                style={{ display : 'block', WebkitUserSelect : 'none', cursor : 'default', fontFamily : this.props.fontFamily, fontWeight : this.props.fontWeight, fontSize : this.props.textSize, color : textColor, pointerEvents : this.props.disabled ? 'none' : 'auto' }}>
                {this.props.children}
            </span>
        );
    }
}

Text.propTypes = {
    className  : React.PropTypes.string,
    fontFamily : React.PropTypes.string,
    fontWeight : React.PropTypes.oneOfType([ React.PropTypes.string, React.PropTypes.number ]),
    textSize   : React.PropTypes.string,
    textColor  : React.PropTypes.string,
    theme      : React.PropTypes.oneOf([ 'light', 'dark' ]),
    disabled   : React.PropTypes.bool
};

Text.defaultProps = {
    className : 'Text',
    theme     : 'light',
    disabled  : false
};

module.exports = Text;
