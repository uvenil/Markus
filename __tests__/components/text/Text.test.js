'use strict';

import React from 'react';
import Renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import Text from '../../../app/components/text/Text.jsx';

it('renders correctly', () => {
    const text = Renderer.create(
        <MuiThemeProvider muiTheme={getMuiTheme(lightBaseTheme)}>
            <Text style={{ fontWeight : 'bold' }}>Hello, World!</Text>
        </MuiThemeProvider>
    ).toJSON();

    expect(text).toMatchSnapshot();
});
