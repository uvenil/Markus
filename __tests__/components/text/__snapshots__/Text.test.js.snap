// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className={undefined}
  onMouseDown={[Function]}
  style={
    Object {
      "WebkitUserSelect": "none",
      "color": "rgba(0, 0, 0, 0.87)",
      "cursor": "default",
      "display": "block",
      "fontWeight": "bold",
      "pointerEvents": "auto",
    }
  }
>
  Hello, World!
</span>
`;
